pkg/controller/stackconfigpolicy/controller.go

```diff
@@ -757,3 +851,265 @@ func (r *ReconcileStackConfigPolicy) addDynamicWatchesOnAdditionalSecretMounts(p
+
+	// Check if Kibana settings are empty
+	kbEmpty := (kb.Config == nil || len(kb.Config.Data) == 0)
+
+	return esEmpty && kbEmpty
+}
+
+// elasticsearchSettingsConflict checks if two policies have conflicting Elasticsearch settings
+func (r *ReconcileStackConfigPolicy) elasticsearchSettingsConflict(policy1, policy2 *policyv1alpha1.StackConfigPolicy) bool {
+	es1 := &policy1.Spec.Elasticsearch
+	es2 := &policy2.Spec.Elasticsearch
+
+	// Check each type of setting for key conflicts
+	if r.configsConflict(es1.ClusterSettings, es2.ClusterSettings) {
+		return true
+	}
+	if r.configsConflict(es1.SnapshotRepositories, es2.SnapshotRepositories) {
+		return true
+	}
+	if r.configsConflict(es1.SnapshotLifecyclePolicies, es2.SnapshotLifecyclePolicies) {
+		return true
+	}
+	if r.configsConflict(es1.SecurityRoleMappings, es2.SecurityRoleMappings) {
+		return true
+	}
+	if r.configsConflict(es1.IndexLifecyclePolicies, es2.IndexLifecyclePolicies) {
+		return true
+	}
+	if r.configsConflict(es1.IngestPipelines, es2.IngestPipelines) {
+		return true
+	}
+	if r.configsConflict(es1.IndexTemplates.ComponentTemplates, es2.IndexTemplates.ComponentTemplates) {
+		return true
+	}
+	if r.configsConflict(es1.IndexTemplates.ComposableIndexTemplates, es2.IndexTemplates.ComposableIndexTemplates) {
+		return true
+	}
+	if r.configsConflict(es1.Config, es2.Config) {
+		return true
+	}
+
+	// Check secret mounts for path conflicts
+	return r.secretMountsConflict(es1.SecretMounts, es2.SecretMounts)
+}
+
+// kibanaSettingsConflict checks if two policies have conflicting Kibana settings
+func (r *ReconcileStackConfigPolicy) kibanaSettingsConflict(policy1, policy2 *policyv1alpha1.StackConfigPolicy) bool {
+	kb1 := &policy1.Spec.Kibana
+	kb2 := &policy2.Spec.Kibana
+
+	return r.configsConflict(kb1.Config, kb2.Config)
+}
+
+// configsConflict checks if two Config objects have overlapping keys
+func (r *ReconcileStackConfigPolicy) configsConflict(config1, config2 *commonv1.Config) bool {
+	if config1 == nil || config2 == nil || config1.Data == nil || config2.Data == nil {
+		return false
+	}
+
+	// Check if there are any common keys
+	for key := range config1.Data {
+		if _, exists := config2.Data[key]; exists {
+			return true
+		}
+	}
+
+	return false
+}
+
+// secretMountsConflict checks if two sets of secret mounts have overlapping mount paths
+func (r *ReconcileStackConfigPolicy) secretMountsConflict(mounts1, mounts2 []policyv1alpha1.SecretMount) bool {
+	if len(mounts1) == 0 || len(mounts2) == 0 {
+		return false
+	}
+
+	paths1 := make(map[string]bool)
+	for _, mount := range mounts1 {
+		paths1[mount.MountPath] = true
+	}
+
+	for _, mount := range mounts2 {
+		if paths1[mount.MountPath] {
+			return true
+		}
+	}
+
+	return false
+}
+
+// policiesCouldOverlap checks if two policies could potentially target the same resources
+func (r *ReconcileStackConfigPolicy) policiesCouldOverlap(policy1, policy2 *policyv1alpha1.StackConfigPolicy, policy1Selector labels.Selector) bool {
```

Comment:
Bit awkward to have the policy selector passed in here. I I think would probably make these receiver methods of StackConfigPolicy instead.